pkg/recipes/terraform/module.go

```diff
@@ -73,9 +75,32 @@ func inspectModule(workingDir string, recipe *recipes.EnvironmentDefinition) (*m
 		result.ContextVarExists = true
 	}
 
-	// Extract the list of required providers.
-	for providerName := range mod.RequiredProviders {
-		result.RequiredProviders = append(result.RequiredProviders, providerName)
+	// Extract the details of required providers.
+	for k, v := range mod.RequiredProviders {
+		requiredprovider := &config.RequiredProviderInfo{}
+
+		if v.Source != "" {
+			requiredprovider.Source = v.Source
+		}
```

Comment:
Here we can just do `requiredprovider.Source = v.Source` because Source is already an empty string based on the definition above in line 80: `requiredprovider := &config.RequiredProviderInfo{}`.